worlds/factorio/data/mod_template/control.lua

```diff
@@ -132,6 +132,9 @@ function fisher_yates_shuffle(tbl)
     return tbl
 end
 
+-- Handle the pathfinding result of teleport traps
+script.on_event(defines.events.on_script_path_request_finished, handle_teleport_attempt)
```

Comment:
This event handler is within the chunk_shuffle if block. Meaning that teleport traps aren't handled outside of chunk shuffle.